.github/workflows/ci.yml

```
@@ -1,91 +1,162 @@
+    - name: Run ingest type checking
+      run: make ingest-type-check
+      continue-on-error: true  # Allow mypy to fail for now
+    
+    - name: Run ingest security scanning
+      run: make ingest-security
+    
+    - name: Run ingest tests
+      run: make ingest-test
+    
+    - name: Upload ingest coverage to Codecov
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)